results/intfloat__multilingual-e5-small/LEMBPasskeyRetrieval.json

```diff
@@ -0,0 +1,43 @@
+{
+  "dataset_revision": "6e346642246bfb4928c560ee08640dc84d074e8c",
+  "mteb_dataset_name": "LEMBPasskeyRetrieval",
+  "mteb_version": "1.6.11",
+  "test": {
+    "evaluation_time": 1.73,
+    "map_at_1": 1.0,
+    "map_at_10": 1.0,
+    "map_at_100": 1.0,
+    "map_at_1000": 1.0,
+    "map_at_20": 1.0,
+    "map_at_3": 1.0,
+    "map_at_5": 1.0,
+    "mrr_at_1": 1.0,
+    "mrr_at_10": 1.0,
+    "mrr_at_100": 1.0,
+    "mrr_at_1000": 1.0,
+    "mrr_at_20": 1.0,
+    "mrr_at_3": 1.0,
+    "mrr_at_5": 1.0,
+    "ndcg_at_1": 1.0,
+    "ndcg_at_10": 1.0,
+    "ndcg_at_100": 1.0,
+    "ndcg_at_1000": 1.0,
+    "ndcg_at_20": 1.0,
+    "ndcg_at_3": 1.0,
+    "ndcg_at_5": 1.0,
```

Comment:
Perfect scores?

Reply:
yes, it is expected. The passkey and needle test have various evaluation lengths: {256,512,1024,2048,4096,8192,16384,32768}. (That's why we need an additional `context length` parameter) This is designed to measure the effective context window length of the embedding models. The reported scores are from 512 context length. Since embedding models typically have a context length of 512 and above, it's not strange for the perfect score here. After all, passkey retrieval itself is very easy. Needle retrieval is a little bit harder, so we did not observe 100% accuracy here

Reply:
Ahh thanks for clarifying. This seems like they should have their own task subtypes.

A potential solution could simply be to include an equal percentage of each size. Then models with higher context length would perform better. An alternative model. Another option would also be to use the splits to differentiate. So e.g. "test_256", "test_512" etc.?

Reply:
Actually it also bothers me in implementation. The first solution by *include an equal percentage of each size* may not work, as I do not want the candidate documents for each length to be mixed together. As for the second choice, I have used `split` to specify `corpus`, `queries` and `qrels`. As a result, I cannot figure out a better way then passing the `context_length` parameter to do the filtering.

Reply:
Well the "test" split seems to be used in the data structure (not needed on HF), so splitting it up during the  the dataset_transform() seems like a valid approach.

Reply:
make sense, I'll try it. Really appreciate your help.

Reply:
No worries, sorry for the long review period, but this seems like a good thing to get right early on in the process.

Reply:
Hi, I have changed the code to use `test_256`, `test_512`, ... instead of the context length. Looks much better now, really appreciate your reminder! Results are also updated. Would you like to take a look? By the way, we have made our repo publicly available here: https://github.com/dwzhu-pku/LongEmbed. Welcome to take a look!